utility/include/isobus/utility/thread_synchronization.hpp

```diff
@@ -25,9 +26,76 @@ namespace isobus
 /// @brief Disabled LOCK_GUARD macro since threads are disabled.
 #define LOCK_GUARD(type, x)
 
+/// @brief A template class for a queue, since threads are disabled this is a simple queue.
+/// @tparam T The item type for the queue.
+template<typename T>
+class LockFreeQueue
+{
+public:
+	/// @brief Constructor for the lock free queue.
+	explicit LockFreeQueue(std::size_t) {}
+
+	/// @brief Push an item to the queue.
+	/// @param item The item to push to the queue.
+	/// @return Simply returns true, since this version of the queue is not limited in size.
+	bool push(const T &item)
+	{
+		queue.push(item);
+		return true;
+	}
+
+	/// @brief Peek at the next item in the queue.
+	/// @param item The item to peek at in the queue.
+	/// @return True if the item was peeked at in the queue, false if the queue is empty.
+	bool peek(T &item)
+	{
+		if (queue.empty())
+		{
+			return false;
+		}
+
+		item = queue.front();
+		return true;
+	}
+
+	/// @brief Pop an item from the queue.
+	/// @return True if the item was popped from the queue, false if the queue is empty.
+	bool pop()
+	{
+		if (queue.empty())
+		{
+			return false;
+		}
+
+		queue.pop();
+		return true;
+	}
+
+	/// @brief Check if the queue is full.
+	/// @return Always returns false, since this version of the queue is not limited in size.
+	bool is_full() const
+	{
+		return false;
+	}
+
+	/// @brief Clear the queue.
+	void clear()
+	{
+		while (!queue.empty())
+		{
+			queue.pop();
```

Comment:
I think this might actually be more efficient, though it probably doesn't matter

`queue = { };`